Subject: [PATCH] don't use fortify-headers on netlink test sources
From: A. Wilcox <AWilcox@Wilcox-Tech.com>

We can't use fortify-headers on netlink tests because it tests what happens
when a buffer overrun occurs.

--- strace-4.18/tests/netlink_protocol.c.old	2017-07-05 07:08:09.000000000 +0000
+++ strace-4.18/tests/netlink_protocol.c	2017-08-17 01:09:45.822502012 +0000
@@ -28,6 +28,8 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#define _FORTIFY_SOURCE 0
+
 #include "tests.h"
 
 #ifdef HAVE_SYS_XATTR_H
--- strace-4.18/tests/netlink_sock_diag.c.old	2017-07-05 07:08:09.000000000 +0000
+++ strace-4.18/tests/netlink_sock_diag.c	2017-08-17 01:10:00.935807300 +0000
@@ -27,6 +27,8 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#define _FORTIFY_SOURCE 0
+
 #include "tests.h"
 #include <stdio.h>
 #include <string.h>
--- strace-4.18/tests/nlattr.c.old	2017-07-05 07:08:09.000000000 +0000
+++ strace-4.18/tests/nlattr.c	2017-08-17 01:10:11.862453682 +0000
@@ -28,6 +28,8 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#define _FORTIFY_SOURCE 0
+
 #include "tests.h"
 
 #include <stdio.h>
